### PR TITLE
Add reducible docs 

### PR DESCRIPTION
**Pull request by LukaJCB**
Looking for some early feedback on this, still mostly unstructured :)

### CODE REVIEW

File: docs/src/main/tut/typeclasses/reducible.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+source: "core/src/main/scala/cats/Reducible.scala"
+scaladoc: "#cats.Reducible"
+---
+
+# Reducible
+
+`Reducible` extends the `Foldable` type class with additional `reduce` methods.
+
+You may have come by one of the `reduce`, `reduceLeft` or `reduceOption` defined in Scala's standard collections.
+`Reducible` offers exactly these methods with the guarantee, that the collection won't throw an exception at runtime, without having to reduce to an `Option`.
```
Comment (ceedubs) [inline docs/src/main/tut/typeclasses/reducible.md]: Minor: I don't think that you need a comma after "guarantee".

Comment (ceedubs) [reply docs/src/main/tut/typeclasses/reducible.md]: I guess that years of Java have made me paranoid. Maybe "the collection won't throw an exception at runtime" is a bit of a strong guarantee. Maybe we should just say that these methods won't throw an exception due to a collection being empty? :)

**Comment by kailuowang** [comment]
Seems history got messed up?

**Comment by LukaJCB** [comment]
Fixed!

**Comment by kailuowang** [comment]
Sorry I dropped ball on this one. @LukaJCB for some reason Github conflict merge tool is not working for us. Do yo mind update this one and merge?

**Comment by LukaJCB** [comment]
I think it's fine actually, no? You guys just need to re-review to merge :)

**Pull request merged**